scala3doc/test/dotty/dokka/ExternalLocationProviderTest.scala

```diff
@@ -0,0 +1,76 @@
+package dotty.dokka
+
+import org.jetbrains.dokka.pages.ContentPage
+import org.jetbrains.dokka.pages.PageNode
+import org.jetbrains.dokka.pages.RootPageNode
+import org.jetbrains.dokka.pages.ModulePage
+import org.jetbrains.dokka.pages.ClasslikePageNode
+import org.jetbrains.dokka.model.DPackage
+import org.jetbrains.dokka.plugability.DokkaContext
+import org.jetbrains.dokka.base.resolvers.external._
+import org.jetbrains.dokka.base.resolvers.shared.{ExternalDocumentation => ED, _}
+import org.jetbrains.dokka.base.resolvers.local._
+import org.jetbrains.dokka.model.DisplaySourceSet
+import dotty.dokka.withNoOrigin
+import dotty.dokka.tasty._
+
+import scala.collection.JavaConverters._
+import java.nio.file.Paths
+import java.nio.file.Path
+import scala.util.matching._
+import dotty.dokka.model.api._
+import java.net.URL
+import org.junit.{Test}
+import org.junit.Assert._
+
+import scala.quoted._
+
+class ExternalLocationProviderTest:
+  def createExternalLocationProvider(docURL: String, ext: String, kind: DocumentationKind) = {
+    val emptyExtDoc = ED(
+          URL(docURL),
+          PackageList(
+            RecognizedLinkFormat.Javadoc1, JSet(), JMap(), URL(docURL)
+          )
+        )
+    ScalaExternalLocationProvider(emptyExtDoc, ext, kind)
+  }
+
+  def testResolvedLinks(provider: ScalaExternalLocationProvider, testcases: List[(DRI, String)]) = testcases.foreach {
+    case (dri, expect) => assertEquals(provider.resolve(dri), expect)
+  }
+
+  @Test
+  def javadocExternalLocationProviderTest(): Unit = {
+    val provider = createExternalLocationProvider("https://docs.oracle.com/javase/8/docs/api/", ".html", DocumentationKind.Javadoc)
+    val testcases = List(
+      (DRI("java.util.Map$$Entry"), "https://docs.oracle.com/javase/8/docs/api/java/util/Map.Entry.html"),
```

Comment:
shouldn't we add a tests for method base `DRI`?

Reply:
I'm gonna make task for that because linking to methods (anchors) are still TODO

Reply:
https://github.com/lampepfl/scala3doc/issues/263